src/proc-service.cc

```diff
@@ -320,6 +320,8 @@ void process_service::handle_exit_status(bp_sys::exit_status exit_status) noexce
     bool was_signalled = exit_status.was_signalled();
     auto current_state = get_state();
 
+    this->exit_status = exit_status;
```

Comment:
This should not be necessary

Reply:
test fails without this but I'm going to check `restart = on-failure` in production environment.

Reply:
I designed a little test for this new feature as below:
(turnstile runs my branch of dinit)
script service:
```
type = process
command = /tmp/script.sh
restart = on-failure
restart-limit-count = 12
log-type = buffer
```
script source:
```
#!/bin/sh

if ! grep "1" /tmp/script.mem; then
    echo "1" >> /tmp/script.mem
    exit 1
fi

if ! grep "2" /tmp/script.mem; then
    echo "2" >> /tmp/script.mem
    exit 2
fi

exit 0
```
run:
```
[mobin@Mobin-Mehdi-PC ~]$ echo "" > /tmp/script.mem
[mobin@Mobin-Mehdi-PC ~]$ dinitctl start script
Service 'script' started.
[mobin@Mobin-Mehdi-PC ~]$ dinitctl catlog script
1
(dinit: note: service restarted)
1
2
[mobin@Mobin-Mehdi-PC ~]$ echo "" > /tmp/script.mem
[mobin@Mobin-Mehdi-PC ~]$ nano ~/.config/dinit.d/script # Changing restart to true
[mobin@Mobin-Mehdi-PC ~]$ dinitctl reload script
Service 'script' reloaded.
[mobin@Mobin-Mehdi-PC ~]$ dinitctl start script
Service 'script' started.
[mobin@Mobin-Mehdi-PC ~]$ dinitctl catlog script
1
(dinit: note: service restarted)
1
2
(dinit: note: service restarted)
(dinit: note: service restarted)
1
(dinit: note: service restarted)
1
2
(dinit: note: service restarted)
1
2
(dinit: note: service restarted)
1
2
(dinit: note: service restarted)
1
2
(dinit: note: service restarted)
1
2
(dinit: note: service restarted)
1
2
(dinit: note: service restarted)
1
2
(dinit: note: service restarted)
1
2
(dinit: note: service restarted)
1
2
(dinit: note: service restarted)
1
2
(dinit: note: service restarted)
1
2
```

Looks like it does work in production environment but not in test environment. @davmac314 Do I should skip test for now? I think exit_status handling needs major refactor.

Reply:
> Do I should skip test for now?

As I said already, we can skip the test for now.

> I think exit_status handling needs major refactor.

I think you are jumping to conclusions here, it needs a small rework in my assessment.

Reply:
test is commented out and I yeah, It's already decent

Reply:
Tests are fixed.